### PR TITLE
Applications+Libraries: Move to `LibFileSystem`

### PR DESCRIPTION
**Pull request by cammo1123**
Part one of many in the process of moving to `LibFileSystem`. The only remaining usages of `DeprecatedFile` is either using the `open/construct` functions, or inside `LibCore` as it creates a Cyclic import.

### CODE REVIEW

**Comment by cammo1123** [comment]
CI's just discovered a leak in real_path. Turning into a draft until I can find a fix

File: Userland/Applications/FileManager/DirectoryView.cpp [hunk 1]
```diff
@@ -406,18 +405,19 @@ void DirectoryView::add_path_to_history(DeprecatedString path)
...
-    auto real_path = Core::DeprecatedFile::real_path_for(path);
-    if (real_path.is_null() || !FileSystem::is_directory(path))
+    auto real_path_or_error = FileSystem::real_path(path);
+    if (real_path_or_error.is_error() || !FileSystem::is_directory(path))
         return false;
 
-    if (chdir(real_path.characters()) < 0) {
+    auto real_path = real_path_or_error.release_value();
+    if (Core::System::chdir(real_path).is_error())
         perror("chdir");
```
Comment (linusg) [inline Userland/Applications/FileManager/DirectoryView.cpp:414]: Pre-existing bug, but `perror` only prints an error - we also need to return false here!

File: Userland/Applications/FileManager/main.cpp [hunk 2]
```diff
@@ -142,14 +141,14 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
     }
 
-    if (initial_location.is_empty())
-        initial_location = Core::DeprecatedFile::current_working_directory();
+    if (auto error_or = FileSystem::current_working_directory(); initial_location.is_empty() && !error_or.is_error())
+        initial_location = error_or.release_value().bytes_as_string_view();
 
     if (initial_location.is_empty())
-        initial_location = Core::StandardPaths::home_directory();
+        initial_location = Core::StandardPaths::home_directory().view();
```
Comment (linusg) [inline Userland/Applications/FileManager/main.cpp:148]: UAF, the `DeprecatedString` you're pointing into disappears after this line.

**Review by linusg — CHANGES_REQUESTED** [review]
Missing backtick in second commit message

File: Userland/Applications/FileManager/main.cpp [hunk 3]
```diff
@@ -109,8 +108,8 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     LexicalPath path(initial_location);
     if (!initial_location.is_empty()) {
-        if (!ignore_path_resolution)
-            initial_location = Core::DeprecatedFile::real_path_for(initial_location);
+        if (auto error_or_path = FileSystem::real_path(initial_location); !ignore_path_resolution && !error_or_path.is_error())
+            initial_location = error_or_path.release_value().bytes_as_string_view();
```
Comment (linusg) [inline Userland/Applications/FileManager/main.cpp:112]: Please be explicit here and use to_deprecated_string

**Comment by linusg** [comment]
This has conflicts 😅

**Pull request merged**